dev/tools/controllerbuilder/template/controller/externalresource.go

```diff
@@ -59,12 +73,12 @@ func asID(externalRef string) (*{{.Kind}}Identity, error) {
 	tokens := strings.Split(path, "/")
 
 	// TODO(user): Confirm the format of your resources, and verify it like the example below
-	if len(tokens) != 6 || tokens[0] != "projects" || tokens[2] != "locations" || tokens[4] != "{{.ProtoResource}}s" {
-		return nil, fmt.Errorf("externalRef should be %s/projects/<project>/locations/<location>/{{.ProtoResource}}s/<{{.ProtoResource}}>, got %s",
+	if len(tokens) != 6 || tokens[0] != "projects" || tokens[2] != "locations" || tokens[4] != "{{.ProtoResource | ToLower}}s" {
```

Comment:
Nit: we should probably have a helper for `{{.ProtoResource | ToLower}}s`, like `{{.URLComponentName}}`

Will be helpful if things don't follow the pattern (or have complicated plurals)!